modules/storage-aws-s3/s3/config.go

```diff
@@ -46,6 +48,13 @@ func (c *config) BucketName() string {
 	return DEFAULT_BUCKET
 }
 
+func (c *config) RootName() string {
```

Comment:
This would be equivalent to  `func (c *config) RootName() string{ return c.root }`

Reply:
Agreed, but I felt that the added verbosity helps convey the idea that this value can be optionally set, and that an empty value here would be expected.

Reply:
Fixed to your suggestion